src/TheRestOfYourLife/bvh.h

```diff
@@ -28,131 +36,82 @@ class bvh_node : public hittable  {
         aabb box;
 };
 
-bool bvh_node::bounding_box(double t0, double t1, aabb& output_box) const {
-    output_box = box;
-    return true;
-}
 
-bool bvh_node::hit(const ray& r, double t_min, double t_max, hit_record& rec) const {
-    if (!box.hit(r, t_min, t_max))
-        return false;
+bool box_compare(const hittable *a, const hittable *b, int axis) {
+    aabb box_a;
+    aabb box_b;
 
-    bool hit_left = left->hit(r, t_min, t_max, rec);
-    bool hit_right = right->hit(r, t_min, hit_left ? rec.t : t_max, rec);
+    if (!a->bounding_box(0,0, box_a) || !b->bounding_box(0,0, box_b))
+        std::cerr << "No bounding box in bvh_node constructor.\n";
 
-    return hit_left || hit_right;
+    return box_a.min().e[axis] < box_b.min().e[axis];
 }
 
-int box_x_compare (const void * a, const void * b) {
-    aabb box_left, box_right;
-    hittable *ah = *(hittable**)a;
-    hittable *bh = *(hittable**)b;
 
-    if (!ah->bounding_box(0,0, box_left) || !bh->bounding_box(0,0, box_right))
-        std::cerr << "no bounding box in bvh_node constructor\n";
+bool box_x_compare (const hittable *a, const hittable *b) { return box_compare(a, b, 0); }
+bool box_y_compare (const hittable *a, const hittable *b) { return box_compare(a, b, 1); }
+bool box_z_compare (const hittable *a, const hittable *b) { return box_compare(a, b, 2); }
 
-    if (box_left.min().x() - box_right.min().x() < 0.0)
-        return -1;
-    else
-        return 1;
-}
 
-int box_y_compare (const void * a, const void * b)
-{
-    aabb box_left, box_right;
-    hittable *ah = *(hittable**)a;
-    hittable *bh = *(hittable**)b;
+bvh_node::bvh_node(
+    std::vector<hittable*> &objects, size_t start, size_t end, double time0, double time1
+) {
+    if (end <= start) {
```

Comment:
Should this be copied over to _TheNextWeek_?

Reply:
_Oif_. Excellent catch.